dialogue-example/src/main/java/com/palantir/dialogue/example/SampleServiceClient.java

```diff
@@ -98,10 +98,12 @@ public String stringToString(String objectId, String header, String body) {
                         .build();
 
                 Call call = channel.createCall(STRING_TO_STRING, request);
-                ListenableFuture<Response> response = Calls.toFuture(call);
+                ListenableFuture<String> response = Futures.transform(
```

Comment:
thoughts on composing the timeouts as a `Channel` implementation?

Reply:
hmm. it wouldn't be so straightforward, because then the async and the blocking client would need different types of channels. (the async client as it stands doesn't do timeouts as callers are in control.) I think I'd revisit that problem when exposing clients that interact with an `Observer`, because we'd then have to think about how to expose the timeout to those observers.

Reply:
(sorry, missed this comment before the last one I posted)

It seems like async clients should still enforce timeouts on the requests/time-to-response, in which case it'd be the same Channel impl?